src/zcl_abapgit_repo_online.clas.abap

```diff
@@ -86,31 +69,40 @@ CLASS zcl_abapgit_repo_online DEFINITION
         !it_objects TYPE zif_abapgit_definitions=>ty_objects_tt
       RAISING
         zcx_abapgit_exception .
+    METHODS fetch_remote
+      RAISING
+        zcx_abapgit_exception .
 ENDCLASS.
 
 
 
 CLASS ZCL_ABAPGIT_REPO_ONLINE IMPLEMENTATION.
 
 
-  METHOD constructor.
-
-    super->constructor( is_data ).
-
-    mv_initialized = abap_false.
+  METHOD fetch_remote.
 
-  ENDMETHOD.
+    DATA: lo_progress TYPE REF TO zcl_abapgit_progress,
+          ls_pull     TYPE zcl_abapgit_git_porcelain=>ty_pull_result.
 
+    IF mv_remote_is_initialized = abap_true.
+      RETURN.
+    ENDIF.
 
-  METHOD deserialize.
+    CREATE OBJECT lo_progress
+      EXPORTING
+        iv_total = 1.
 
-    initialize( ).
 
-    super->deserialize( is_checks ).
+    lo_progress->show( iv_current = 1
+                       iv_text    = 'Fetch remote files' ) ##NO_TEXT.
 
-    reset_status( ).
+    ls_pull = zcl_abapgit_git_porcelain=>pull(
+      iv_url         = get_url( )
+      iv_branch_name = get_branch_name( ) ).
 
-    COMMIT WORK AND WAIT.
+    set_files_remote( ls_pull-files ).
+    set_objects( ls_pull-objects ).
```

Comment:
set_objects was previously only called during push?

Reply:
it was `mt_objects = ls_pull-objects` before. So the same, just unified the assignment.

BTW as we touched this: in fact `mt_objects` just called/modified internally (in `fetch` and `push`). I didn't do it but I though to remove `get/set_objects` at all and address just the attribute directly. What do you think ?